Update the FQN for auto_functionalized HOO.

zhxchen17 opened the pull request:
Summary:
as title.

torch._higher_order_ops.auto_functionlize.auto_functionalized is a Python FQN which should NOT be used to talk to the backends and we should use the standard FQN name torch.ops.higher_order.auto_functionalized instead.

Test Plan: buck test mode/opt caffe2/test:test_export -- -r test_custom_op_auto_functionalize_pre_dispatch

Differential Revision: D60468759

zou3519 commented:
constructing a HigherOrderOperator should probably set the FQN, so we would fix all the problems in one go. But otherwise this is how we handle cond, so go for it.

zhxchen17 committed:
Update the FQN for auto_functionalized HOO. (#132171)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/132171

as title.

torch._higher_order_ops.auto_functionlize.auto_functionalized is a Python FQN which should NOT be used to talk to the backends and we should use the standard FQN name torch.ops.higher_order.auto_functionalized instead.

Test Plan: buck test mode/opt caffe2/test:test_export -- -r test_custom_op_auto_functionalize_pre_dispatch

Differential Revision: D60468759

zhxchen17 commented:
> constructing a HigherOrderOperator should probably set the FQN, so we would fix all the problems in one go. But otherwise this is how we handle cond, so go for it.

@zou3519  yeah I agree that it's better to just do that in the ctor. Not blocked on that but I can make another diff to fix everything in one go.

izaitsevfb commented:
@pytorchbot merge

izaitsevfb commented:
@pytorchbot merge -f 'broken trunk prevents landing'
